### PR TITLE
feature/mqtt-publisher-example

### PR DESCRIPTION
**Pull request by suneetnangia**
Added mqtt outbound example, to be merged once the [feature](https://github.com/fermyon/spin/pull/2287) in Spin is merge. 
Changes are tested on locally built Spin host from the above PR.

### CODE REVIEW

File: wit/deps/spin@unversioned/mqtt-types.wit [hunk 1]
```diff
@@ -0,0 +1,17 @@
+interface mqtt-types {
```
Comment (fibonacci1729) [inline wit/deps/spin@unversioned/mqtt-types.wit:1]: I'm not sure adding the `mqtt` interface to `spin@unversioned` is necessary here. The `spin@unversioned` package is here to maintain backwards compatibility with Spin V1 components. New features should be added to root level `wit` package as you have done below.

**Review by fibonacci1729 — APPROVED** [review]
Aside from the one comment this looks great!

File: wit/deps/spin@unversioned/mqtt-types.wit (hunk 1 above)
Comment (suneetnangia) [reply wit/deps/spin@unversioned/mqtt-types.wit:1]: Thanks, will update.

File: wit/deps/spin@unversioned/world.wit [hunk 2]
```diff
@@ -22,7 +22,7 @@ world platform {
   import http;
   import postgres;
   import mysql;
-  import sqlite;
+  import sqlite;
```
Comment (fibonacci1729) [inline wit/deps/spin@unversioned/world.wit:25]:
```suggestion
  import sqlite;
```

**Pull request merged**